python/lsst/ip/isr/calibType.py

```diff
@@ -362,17 +363,26 @@ def search(haystack, needles):
                 raise RuntimeError(f"Incorrect calibration supplied.  Expected {self._OBSTYPE}, "
                                    f"found {metadata['OBSTYPE']}")
 
-        self._instrument = search(dictionary, ["INSTRUME", "instrument"])
-        self._raftName = search(dictionary, ["RAFTNAME"])
-        self._slotName = search(dictionary, ["SLOTNAME"])
-        self._detectorId = search(dictionary, ["DETECTOR", "detectorId"])
-        self._detectorName = search(dictionary, ["DET_NAME", "DETECTOR_NAME", "detectorName"])
-        self._detectorSerial = search(dictionary, ["DET_SER", "DETECTOR_SERIAL", "detectorSerial"])
-        self._filter = search(dictionary, ["FILTER", "filterName"])
-        self._calibId = search(dictionary, ["CALIB_ID"])
-        self._seqfile = search(dictionary, ["SEQFILE"])
-        self._seqname = search(dictionary, ["SEQNAME"])
-        self._seqcksum = search(dictionary, ["SEQCKSUM"])
+        if (value := search(dictionary, ["INSTRUME", "instrument"])) is not None:
```

Comment:
At least half of these can come straight out of ObservationInfo if you want to consider that as a future way of not being directly dependent on every FITS header being exactly what you need.